### PR TITLE
Rolling updater should allow progress to be logged / detected

### PR DESCRIPTION
**Pull request by smarterclayton**
Allows callers to terminate early as well as print useful messages
during the rollout. Use a crude percentage metric for summarization.

### CODE REVIEW

**Comment by kargakis** [comment]
lgtm but
```
pkg/kubectl/rolling_updater.go:239: cannot use (goal - progress) * 100 / goal (type int32) as type int in assignment
```

**Comment by kargakis** [comment]
cc: @kubernetes/kubectl

**Comment by bgrant0607** [comment]
cc @janetkuo

File: pkg/kubectl/rolling_updater.go [hunk 1]
```diff
@@ -217,6 +221,26 @@ func (r *RollingUpdater) Update(config *RollingUpdaterConfig) error {
...
+		if config.OnProgress == nil {
+			return nil
+		}
+		progress := desired - newRc.Spec.Replicas
+		if progress < 0 {
+			progress = -progress
+		}
+		percentage := 100
+		if !complete && goal > 0 {
+			percentage = (goal - progress) * 100 / goal
```
Comment (bgrant0607) [inline pkg/kubectl/rolling_updater.go]: Is it possible for the goal to be 0?

**Comment by smarterclayton** [comment]
@k8s-bot test this issue #IGNORE

**Comment by bprashanth** [comment]
@k8s-bot test this github issue: #25629

File: pkg/kubectl/rolling_updater.go (hunk 1 above)
Comment (kargakis) [inline pkg/kubectl/rolling_updater.go]: It seems so...

Comment (smarterclayton) [inline pkg/kubectl/rolling_updater.go]: Yes, it's possible the function can get called with desired - newRc.Spec.Replicas in a few cases. In which case you are considered to be at 100%, i.e. at goal.

**Comment by smarterclayton** [comment]
@k8s-bot test this github issue: #25629

**Comment by smarterclayton** [comment]
@k8s-bot test this github issue: #25629

**Comment by kargakis** [comment]
LGTM'd

**Pull request merged**